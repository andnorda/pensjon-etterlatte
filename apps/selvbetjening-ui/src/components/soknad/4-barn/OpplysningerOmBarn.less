@import "../../../App.less";

.barnekort-wrapper {
    display: flex;
    flex-flow: row wrap;
    justify-content: flex-start;
    margin: 0 auto;
    width: 620px;

    .barnekort {
        .typo-normal,
        .typo-element {
            font-size: 18px;
        }
        margin: 0 1rem 1rem 0;
        width: 276px;
        background-color: #e7e9e9;

        border-bottom-left-radius: 5px;
        border-bottom-right-radius: 5px;

        &.legg-til {
            height: 230px;
        }
        &__header {
            box-sizing: border-box;
            height: 128px;
            background-color: #4d3e55;
            border-top-right-radius: 5px;
            border-top-left-radius: 5px;
            border-bottom: 4px solid #826ba1;
            display: flex;
            align-items: flex-end;

            img {
                display: block;
                margin: 0 auto;
            }
        }

        &__informasjonsboks {
            padding: 2rem 2rem;
            .informasjonsboks-innhold {
                height: 100%;
                display: flex;
                flex-direction: column;
                justify-content: flex-start;

                .informasjonselement {
                    margin: 10px 0 10px 0;
                }
            }
        }

        &__endre-barnekort {
            border: none;
            text-align: left;
            padding: inherit;
        }

        .legg-til-barn-kort {
            border-radius: 5px;
            height: 200px;

            .typo-element {
                padding-bottom: 2rem;
            }
        }
    }
}
