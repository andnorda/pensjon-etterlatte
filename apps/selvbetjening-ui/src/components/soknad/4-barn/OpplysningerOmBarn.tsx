import React, { useState } from "react";
import "../../../App.less";
import "../../felles/Infokort.less";
import ikon from "../../../assets/ikoner/barn1.svg";
import { RadioPanelGruppe, SkjemaGruppe } from "nav-frontend-skjema";
import { Normaltekst, Systemtittel, Undertittel } from "nav-frontend-typografi";
import { Hovedknapp, Knapp } from "nav-frontend-knapper";
import SoknadSteg from "../../../typer/SoknadSteg";
import ToValgRadio from "../../felles/ToValgRadio";
import { default as Modal } from "nav-frontend-modal";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IBarn } from "../../../typer/person";
import { SoeknadActionTypes } from "../../../context/soknad/soknad";
import TekstInput from "../../felles/TekstInput";
import { useTranslation } from "react-i18next";

Modal.setAppElement("#root");

const OpplysningerOmBarn: SoknadSteg = () => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    // Modal
    const [isOpen, setIsOpen] = useState(false);

    const tomtElement: IBarn = {
        fornavn: "",
        etternavn: "",
        foedselsnummer: "",
        foreldre: "",
        bosattUtland: undefined,
        statsborgerskap: "",
        land: "",
    };

    const [barn, setBarn] = useState(tomtElement);

    const leggTilOgLukk = () => {
        leggTil();
        setIsOpen(false);
        setBarn(tomtElement);
    };

    const leggTil = () => {
        dispatch({ type: SoeknadActionTypes.LEGG_TIL_BARN, payload: barn });
        setBarn(tomtElement);
    };

    const lukkModalvindu = () => {
        setIsOpen(false);
        setBarn(tomtElement);
    };

    return (
        <>
            {/* Steg 4 */}
            <Systemtittel>{t("opplysningerOmBarn.tittel")}</Systemtittel>

            <div className={"infokort-wrapper"}>
                {state.opplysningerOmBarn?.map((barn) => {
                    return (
                        <div className={"infokort"} key={barn.foedselsnummer}>
                            <div className={"infokort__header"}>
                                <img alt="barn" className="barneikon" src={ikon} />
                            </div>
                            <div className={"infokort__informasjonsboks"}>
                                <div className={"informasjonsboks-innhold"}>
                                    <Undertittel tag="h3">
                                        {barn.fornavn} {barn.etternavn}
                                    </Undertittel>
                                </div>
                                <div className="informasjonselement">
                                    <Normaltekst>{barn.foedselsnummer}</Normaltekst>
                                    <Normaltekst>{barn.foreldre}</Normaltekst>
                                    <Normaltekst>
                                        {t("opplysningerOmBarn.bosattUtland")}: {barn.bosattUtland}
                                    </Normaltekst>
                                </div>
                            </div>
                        </div>
                    );
                })}

                <div className={"infokort"}>
                    <div className={"infokort__header gjennomsiktig"}>
                        <img alt="barn" className="barneikon" src={ikon} />
                    </div>
                    <div className={"infokort__informasjonsboks"}>
                        <div className={"informasjonsboks-innhold"}>
                            <Knapp onClick={() => setIsOpen(true)}>{t("knapp.leggTilBarn")}</Knapp>
                        </div>
                    </div>
                </div>
            </div>

            <Modal
                isOpen={isOpen}
                onRequestClose={lukkModalvindu}
                closeButton={true}
                contentLabel={t("opplysningerOmBarn.modalContentLabel")}
            >
                <div style={{ padding: "2rem 2.5rem" }}>
                    <SkjemaGruppe>
                        {/* sjekkboks for INGEN BARN */}

                        <TekstInput
                            label={t("felles.fornavn")}
                            value={barn.fornavn}
                            onChange={(fornavn) => setBarn({ ...barn, fornavn })}
                        />

                        <TekstInput
                            label={t("felles.etternavn")}
                            value={barn.etternavn}
                            onChange={(etternavn) => setBarn({ ...barn, etternavn })}
                        />

                        <TekstInput
                            label={t("felles.fnr")}
                            value={barn.foedselsnummer}
                            onChange={(foedselsnummer) => setBarn({ ...barn, foedselsnummer })}
                        />

                        <RadioPanelGruppe
                            name={"barn-opphav"}
                            radios={[
                                { label: t("opplysningerOmBarn.fellesbarnMedAvdoed"), value: "Fellesbarn m/avdøde" },
                                { label: t("opplysningerOmBarn.avdoedesSaerkullsbarn"), value: "Avdødes særkullsbarn" },
                                { label: t("opplysningerOmBarn.egneSaerkullsbarn"), value: "Egne særkullsbarn" },
                            ]}
                            checked={barn.foreldre}
                            onChange={(e) => setBarn({ ...barn, foreldre: (e.target as HTMLInputElement).value })}
                        />

                        <ToValgRadio
                            checked={barn.bosattUtland}
                            label={t("opplysningerOmBarn.borUtenlands")}
                            onChange={(valgtSvar) => setBarn({ ...barn, bosattUtland: valgtSvar })}
                        >
                            <TekstInput
                                label={t("felles.statsborgerskap")}
                                value={barn.statsborgerskap}
                                onChange={(statsborgerskap) =>
                                    setBarn({
                                        ...barn,
                                        statsborgerskap,
                                    })
                                }
                            />

                            <TekstInput
                                label={t("felles.land")}
                                value={barn.land}
                                onChange={(land) =>
                                    setBarn({
                                        ...barn,
                                        land,
                                    })
                                }
                            />
                        </ToValgRadio>

                        <section className={"navigasjon-rad"}>
                            <Knapp onClick={leggTilOgLukk}>{t("knapp.leggTilOgLukk")}</Knapp>
                            <Hovedknapp onClick={leggTil}>{t("knapp.leggTil")}</Hovedknapp>
                        </section>
                    </SkjemaGruppe>
                </div>
            </Modal>
        </>
    );
};

export default OpplysningerOmBarn;
