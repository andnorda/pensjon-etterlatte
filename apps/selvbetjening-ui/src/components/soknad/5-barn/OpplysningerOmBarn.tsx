import React, { useState } from "react";
import "../../felles/Infokort.scss";
import ikon from "../../../assets/ikoner/barn1.svg";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IBarn, IOmBarn } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import BarnInfokort from "./BarnInfokort";
import LeggTilBarnSkjema from "./LeggTilBarnSkjema";
import { SkjemaGruppe } from "nav-frontend-skjema";
import { v4 as uuid } from "uuid";
import Navigasjon from "../../felles/Navigasjon";
import { Alert, BodyShort, Button, Modal, Panel, Heading } from "@navikt/ds-react";
import { FieldArrayWithId, FormProvider, useFieldArray, useForm } from "react-hook-form";
import { RHFSpoersmaalRadio } from "../../felles/RHFRadio";
import useEffectOnce from "../../../hooks/useEffectOnce";
import { isEmpty } from "lodash";

if (process.env.NODE_ENV !== "test") Modal.setAppElement!!("#root"); //Denne er ogsÃ¥ definert i Navigasjon. Trenger vi den?

const OpplysningerOmBarn: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const methods = useForm<IOmBarn>({
        defaultValues: state.opplysningerOmBarn || {},
    });

    useEffectOnce(() => {
        methods.reset(state.opplysningerOmBarn);
    }, !isEmpty(state.opplysningerOmBarn));

    const { fields, append, remove } = useFieldArray({
        name: "barn",
        control: methods.control,
    });

    // Modal
    const [isOpen, setIsOpen] = useState(false);

    const leggTilBarn = (data: IBarn) => {
        append(data as any);
        setIsOpen(false);
    };

    const lagre = (data: IOmBarn) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: data });
        neste!!();
    };

    const { handleSubmit } = methods;

    return (
        <FormProvider {...methods}>
            <form>
                <SkjemaGruppe>
                    <Heading size={"medium"} className={"center"}>
                        {t("omBarn.tittel")}
                    </Heading>
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <Panel border>
                        <Alert variant={"info"} className={"navds-alert--inline"}>
                            <BodyShort size={"small"}>{t("omBarn.informasjon")}</BodyShort>
                        </Alert>
                    </Panel>
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <div className={"infokort-wrapper"}>
                        {fields?.map((field: FieldArrayWithId, index: number) => (
                            <BarnInfokort key={uuid()} barn={field as IBarn} index={index} fjern={remove} />
                        ))}

                        <div className={"infokort"}>
                            <div className={"infokort__header gjennomsiktig"}>
                                <img alt="barn" className="barneikon" src={ikon} />
                            </div>
                            <div className={"infokort__informasjonsboks"}>
                                <div className={"informasjonsboks-innhold"}>
                                    <Button variant={"primary"} type={"button"} onClick={() => setIsOpen(true)}>
                                        {t("knapp.leggTilBarn")}
                                    </Button>
                                </div>

                                <BodyShort size={"small"} className={"center mute"}>
                                    {t("omBarn.valgfritt")}
                                </BodyShort>
                            </div>
                        </div>
                    </div>

                    <Modal open={isOpen} onClose={() => setIsOpen(false)} className={"modal"}>
                        <LeggTilBarnSkjema lagre={leggTilBarn} avbryt={() => setIsOpen(false)} />
                    </Modal>
                </SkjemaGruppe>

                <RHFSpoersmaalRadio name={"gravidEllerNyligFoedt"} legend={t("omBarn.gravidEllerNyligFoedt")} />

                <Navigasjon forrige={{ onClick: forrige }} neste={{ onClick: handleSubmit(lagre) }} />
            </form>
        </FormProvider>
    );
};

export default OpplysningerOmBarn;
