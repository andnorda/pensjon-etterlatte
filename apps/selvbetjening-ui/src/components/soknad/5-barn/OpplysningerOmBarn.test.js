import { render } from "@testing-library/react";
import uuid from "uuid";
import lodash from "lodash";
import * as JSutils from "nav-frontend-js-utils";
import OpplysningerOmBarn from "./OpplysningerOmBarn";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

JSutils.guid = jest.fn(() => "123");
describe("Om den avdÃ¸de", () => {
    it("Snapshot", () => {
        const { container } = render(<OpplysningerOmBarn />);
        expect(container).toMatchSnapshot();
    });
});
