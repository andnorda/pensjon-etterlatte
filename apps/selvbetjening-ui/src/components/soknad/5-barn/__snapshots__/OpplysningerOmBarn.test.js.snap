// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om den avd√∏de Snapshot 1`] = `
<div>
  <form>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <h1
        class="center navds-title navds-title--m"
      >
        omBarn.tittel
      </h1>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <div
        class="navds-panel navds-panel--border"
      >
        <div
          class="navds-alert--inline navds-alert navds-alert--info navds-alert--m navds-body-long"
        >
          <span>
            <span
              class="sr-only"
            >
              info-ikon
            </span>
            <svg
              aria-label="info-ikon"
              fill="none"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0zM9 19v-2h2v-5H9v-2h4v7h2v2H9zm3-14a1.5 1.5 0 110 3 1.5 1.5 0 010-3z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <div>
            <p
              class="navds-body-short navds-body--s"
            >
              omBarn.informasjon
            </p>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe"
    >
      <div
        class="infokort-wrapper"
      >
        <div
          class="infokort"
        >
          <div
            class="infokort__header gjennomsiktig"
          >
            <img
              alt="barn"
              class="barneikon"
              src="barn1.svg"
            />
          </div>
          <div
            class="infokort__informasjonsboks"
          >
            <div
              class="informasjonsboks-innhold"
            >
              <button
                class="navds-button navds-button--primary navds-button--m navds-body-short"
                type="button"
              >
                knapp.leggTilBarn
              </button>
            </div>
            <p
              class="center mute navds-body-short navds-body--s"
            >
              omBarn.valgfritt
            </p>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <div
      id="gravidEllerNyligFoedt"
    >
      <fieldset
        aria-invalid="false"
        class="skjemagruppe inputPanelGruppe inline"
      >
        <legend
          class="skjemagruppe__legend"
        >
          omBarn.gravidEllerNyligFoedt
        </legend>
        <div
          class="inputPanelGruppe__inner"
        >
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="gravidEllerNyligFoedt"
              type="radio"
              value="Ja"
            />
            <span
              class="inputPanel__label"
            >
              Ja
            </span>
          </label>
          <label
            class="inputPanel radioPanel"
          >
            <input
              aria-checked="false"
              aria-invalid="false"
              class="inputPanel__field"
              name="gravidEllerNyligFoedt"
              type="radio"
              value="Nei"
            />
            <span
              class="inputPanel__label"
            >
              Nei
            </span>
          </label>
        </div>
        <div
          aria-live="polite"
        />
      </fieldset>
    </div>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-footer"
    >
      <div
        class="navigasjon-rad"
      >
        <button
          class="navds-button navds-button--primary navds-button--m navds-body-short"
          type="button"
        >
          knapp.tilbake
        </button>
        <button
          class="navds-button navds-button--action navds-button--m navds-body-short"
          type="button"
        >
          knapp.neste
        </button>
      </div>
      <div
        aria-live="polite"
      />
    </fieldset>
    <fieldset
      aria-invalid="false"
      class="skjemagruppe navigasjon-rad"
    >
      <button
        class="navds-button navds-button--secondary navds-button--m navds-body-short"
        type="button"
      >
        knapp.avbryt
      </button>
      <div
        aria-live="polite"
      />
    </fieldset>
  </form>
</div>
`;
